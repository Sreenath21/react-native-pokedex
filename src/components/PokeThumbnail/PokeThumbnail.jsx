import React, {useContext} from 'react';
import {View, Text, Image, Pressable} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

import {
  getZeroFilledId,
  capitalize,
  getGradient,
} from '../../utils/helplerFunctions';
import {PokemonContext} from '../../context/PokemonContext';
import styles from './PokeThumbnail.style';

const PokeThumbnail = ({pokemon, navigation}) => {
  const {setPokemon} = useContext(PokemonContext);

  const handlePress = () => {
    setPokemon(pokemon);
    navigation.navigate('Data', {id: pokemon.id, name: pokemon.name});
  };

  return (
    <Pressable testID="card" onPress={handlePress}>
      <View style={styles.thumbnailContainer}>
        <LinearGradient style={styles.gradient} colors={getGradient(pokemon)}>
          <View style={styles.imageContainer}>
            <Image
              style={styles.image}
              source={{
                uri: `https://img.pokemondb.net/sprites/omega-ruby-alpha-sapphire/dex/normal/${pokemon.name}.png`,
              }}
            />
          </View>

          <View style={styles.textContainer}>
            <Text style={styles.name}>{capitalize(pokemon.name)}</Text>
            <Text style={styles.id}>{getZeroFilledId(pokemon.id)}</Text>
          </View>
        </LinearGradient>
      </View>
    </Pressable>
  );
};

export default PokeThumbnail;
