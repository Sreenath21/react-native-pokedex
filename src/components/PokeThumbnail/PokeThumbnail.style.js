import {StyleSheet} from 'react-native';
import COLORS from '../../utils/colors';

export default StyleSheet.create({
  thumbnailContainer: {
    width: 155,
    height: 225,
    borderColor: 'black',
    borderWidth: 1,
    borderStyle: 'dashed',
    borderRadius: 10,
    overflow: 'hidden',
  },
  gradient: {
    flex: 1,
    borderRadius: 10, // check with overflow
  },
  imageContainer: {
    flex: 7,
  },
  image: {
    height: 160,
    width: 160,
    alignSelf: 'center',
    top: 15,
  },
  textContainer: {
    flex: 2,
  },
  name: {
    color: COLORS.primary,
    fontSize: 16,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  id: {
    color: COLORS.primary,
    textAlign: 'center',
    fontSize: 16,
  },
});
