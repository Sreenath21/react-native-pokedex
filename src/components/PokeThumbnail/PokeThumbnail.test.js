import React from 'react';
import PokeThumbnail from './PokeThumbnail';
import {render} from '@testing-library/react-native';
import PokemonContextProvider from '../../context/PokemonContext';

it('PokeThumbnail renders correctly', () => {
  const screen = render(
    <PokeThumbnail
      pokemon={{name: 'bulbasaur', id: 1, types: []}}
      navigation={jest.fn()}
    />,
    {wrapper: PokemonContextProvider},
  );

  expect(screen).toMatchSnapshot();
});
