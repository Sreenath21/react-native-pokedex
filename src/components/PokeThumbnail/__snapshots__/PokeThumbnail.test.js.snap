// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeThumbnail renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="card"
>
  <View
    style={
      {
        "borderColor": "black",
        "borderRadius": 10,
        "borderStyle": "dashed",
        "borderWidth": 1,
        "height": 225,
        "overflow": "hidden",
        "width": 155,
      }
    }
  >
    <BVLinearGradient
      colors={
        [
          undefined,
        ]
      }
      endPoint={
        {
          "x": 0.5,
          "y": 1,
        }
      }
      locations={null}
      startPoint={
        {
          "x": 0.5,
          "y": 0,
        }
      }
      style={
        {
          "borderRadius": 10,
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "flex": 7,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://img.pokemondb.net/sprites/omega-ruby-alpha-sapphire/dex/normal/bulbasaur.png",
            }
          }
          style={
            {
              "alignSelf": "center",
              "height": 160,
              "top": 15,
              "width": 160,
            }
          }
        />
      </View>
      <View
        style={
          {
            "flex": 2,
          }
        }
      >
        <Text
          style={
            {
              "color": "#391363",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Bulbasaur
        </Text>
        <Text
          style={
            {
              "color": "#391363",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          001
        </Text>
      </View>
    </BVLinearGradient>
  </View>
</View>
`;
